### DISCUSSION TITLE
Undertanding `is_complete`

### DISCUSSION THREAD

**Discussion by Rouneq** [kannibalox/pyrosimple#67] [Q&A]
I was playing around with `rtcontrol` to get a better understanding of how to use it. Specifically, I wanted to see if I could limit it to a single newly defined view. Found the `--from-view` option and started testing. First discovered the utility really doesn't like not having _some_ filter. Fine. Thought, "Ok. Let's just filter on completed torrents." Used the command

`rtcontrol --from=freeleech is_completed -o name`

No results. Hmm. After a bit of thinking I used

`rtcontrol --from=freeleech done=100 -o name,is_complete`

Results kicked out as

```
Torrent.1   DONE
Torrent.2   DONE
```

Then tried

`rtcontrol --from=freeleech is_complete=DONE -o name,is_complete`

and all hell broke loose.

`pyrosimple.util.matching.FilterError: Bad boolean value 'DONE' in 'DONE' (expected one of '{'t', 'yes', 'true', '1', 'y'}', or '{'0', 'no', 'false', 'f', 'n'}')`

Tested with

`rtcontrol --from=freeleech is_complete=true -o name,is_complete`

and received duplicate results as above. Just confirming a testing criteria must be provided even for known boolean values.

**Comment by kannibalox** [comment]
> First discovered the utility really doesn't like not having some filter.

Yep, I typically use `//` as a quick shortcut to get all torrents.

> `rtcontrol --from=freeleech is_completed -o name`

If it doesn't have an `=` in it, rtcontrol assumes it's a name matcher, which you can see in the `--debug` output:
`DEBUG:pyrosimple.scripts.rtcontrol.RtorrentControl:Matcher is: name=is_complete`

> `rtcontrol --from=freeleech is_complete=DONE -o name,is_complete`

All core fields have a default formatter, so while "DONE" or "PART" is the way it comes out for that particular command, underneath the hood it's a regular boolean value. You can always use the `raw` formatter to see the underlying value, e.g. `-o is_complete.raw`.

> Just confirming a testing criteria must be provided even for known boolean values.

Correct, all fields (not just booleans) require a value to match against for rtcontrol's filters.

For the record, `done=100` and `is_completed=yes` use different metrics but functionally work the same.
